### PR TITLE
unsafe-recover needs to specify regions by default

### LINKED ISSUE #3092: Unsafe recover should only drop some nodes instead of all down nodes
For example, A region contains six nodes a, b, c, d, e and f, a is leader. If a, b and c are down forever, we need to just delete two nodes instead of all three nodes. It's possible that just a, b, c and d share the committed log. If a, b, c are deleted, d's committed log can be overridden if e or f becomes leader. Note that d may even not know that a has committed the log, so the cluster can run smoothly without knowing that committed logs are lost, which can be prevented in this case.

### PR DESCRIPTION
**Pull request by hicqu**
tikv-ctl shouldn't decide remove stores for which regions. That should be users' duty.
After the PR is merged, we can't call the command without `regions` argument. So there won't be any unexpected behavior.

Close #3092 .

### CODE REVIEW

**Comment by BusyJay** [comment]
This option is confusing. As if the flag is set to false, then the command will simply check all the regions and then either do nothing or panic. Looks like the flag is expected to be set to true always.

**Comment by BusyJay** [comment]
Any updates?

**Comment by hicqu** [comment]
Fixed. You can check the new strategy in `test_remove_failed_stores`.

**Comment by hicqu** [comment]
PTAL @BusyJay .

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -1233,26 +1233,34 @@ fn main() {
...
+                        .arg(
+                            Arg::with_name("stores")
+                                .required(true)
+                                .short("s")
+                                .takes_value(true)
+                                .multiple(true)
+                                .use_delimiter(true)
+                                .require_delimiter(true)
+                                .value_delimiter(",")
+                                .help("failed store id list"),
```
Comment (BusyJay) [inline src/bin/tikv-ctl.rs]: I think it means `stores to be removed instead of just failed stores`.

File: src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -1233,26 +1233,34 @@ fn main() {
...
+                                .takes_value(true)
+                                .short("r")
+                                .multiple(true)
+                                .use_delimiter(true)
+                                .require_delimiter(true)
+                                .value_delimiter(",")
+                                .help("only for these regions"),
+                        )
+                        .arg(
+                            Arg::with_name("for-all-regions")
```
Comment (BusyJay) [inline src/bin/tikv-ctl.rs]: `for-` is redundant.

File: src/bin/tikv-ctl.rs [hunk 3]
```diff
@@ -1233,26 +1233,34 @@ fn main() {
...
+                                .takes_value(true)
+                                .multiple(true)
+                                .use_delimiter(true)
+                                .require_delimiter(true)
+                                .value_delimiter(",")
+                                .help("failed store id list"),
+                        )
+                        .arg(
+                            Arg::with_name("regions")
+                                .required_unless("for-all-regions")
```
Comment (BusyJay) [inline src/bin/tikv-ctl.rs]: Don't they conflict with each other?

**Comment by BusyJay** [comment]
LGTM, docs may need to be updated too.

**Comment by hicqu** [comment]
@AndreMouche PTAL thanks.

**Pull request merged**